community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/idmapping/string/EncodingIdMapper.java

```diff
@@ -574,7 +578,10 @@ public long dataValue( long nodeId )
                 }
 
                 // Potential duplicate
-                Object inputId = collisionValues.get( collisionIndex );
+                // We cast the collision index to an int here. This means that we can't support > int-range
+                // number of collisions. But that's probably alright since the data structures and
+                // actual collisions values for all these collisions wouldn't fit in a heap anyway.
+                Object inputId = collisionValues.get( Utils.safeCastLongToInt( collisionIndex ) );
```

Comment:
We've heard of people using 64 GiB heaps or more to run Neo4j, so I'm not sure this holds in the pathological case, though in practice it might have a very good chance of being safe. I agree we can at least punt on crossing this bridge for now.